server/models/Request.js

```diff
@@ -0,0 +1,24 @@
+import mongoose from "mongoose";
+
+const { Schema, model } = mongoose;
+
+const requestSchema = new mongoose.Schema(
+  {
+    customer: { type: Schema.Types.ObjectId, ref: "User", required: true },
+    provider: { type: Schema.Types.ObjectId, ref: "User", required: true },
+    start: { type: Date, required: true },
+    end: { type: Date, required: true },
+    accepted: { type: Boolean, default: false },
+    declined: { type: Boolean, default: false },
```

Comment:
I think keeping two fields for tracking the status of a booking introduces unnecessary complexity and isn't the most future-proof solution. I'd recommend reducing this to a single field (maybe we can use `enum` for this too!).